pkg/alerts/muting_rules.go

```diff
@@ -0,0 +1,175 @@
+func (a *Alerts) ListMutingRules(accountID int) ([]MutingRule, error) {
+	vars := map[string]interface{}{
+		"accountID": accountID,
+	}
+
+	resp := alertMutingRuleListResponse{}
+
+	if err := a.client.Query(alertsMutingRulesQuery, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return resp.Actor.Account.Alerts.MutingRules, nil
+}
+
+// CreateMutingRule is the mutation to create a muting rule for the given account and input.
+func (a *Alerts) CreateMutingRule(accountID int, rule MutingRuleCreateInput) (*MutingRule, error) {
+	vars := map[string]interface{}{
+		"accountID": accountID,
+		"rule":      rule,
+	}
+
+	resp := alertMutingRuleCreateResponse{}
+
+	if err := a.client.Query(alertsMutingRulesCreate, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.AlertsMutingRuleCreate, nil
+}
+
+// UpdateMutingRule is the mutation to update an existing muting rule.
+func (a *Alerts) UpdateMutingRule(accountID int, ruleID int, rule MutingRuleUpdateInput) (*MutingRule, error) {
+	vars := map[string]interface{}{
+		"accountID": accountID,
+		"ruleID":    ruleID,
+		"rule":      rule,
+	}
+
+	resp := alertMutingRuleUpdateResponse{}
+
+	if err := a.client.Query(alertsMutingRulesUpdate, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.AlertsMutingRuleUpdate, nil
+}
+
+// DeleteMutingRule is the mutation.
+func (a *Alerts) DeleteMutingRule(accountID int, ruleID int) error {
+	vars := map[string]interface{}{
+		"accountID": accountID,
+		"ruleID":    ruleID,
+	}
+
+	resp := alertMutingRuleDeleteResponse{}
+
+	if err := a.client.Query(alertsMutingRuleDelete, vars, &resp); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type alertMutingRuleCreateResponse struct {
+	AlertsMutingRuleCreate MutingRule `json:"alertsMutingRuleCreate"`
+}
+
+type alertMutingRuleUpdateResponse struct {
+	AlertsMutingRuleUpdate MutingRule `json:"alertsMutingRuleUpdate"`
+}
+
+type alertMutingRuleDeleteResponse struct {
+	AlertsMutingRuleDelete struct {
+		ID string `json:"id"`
+	} `json:"alertsMutingRuleDelete"`
+}
+
+type alertMutingRuleListResponse struct {
+	Actor struct {
+		Account struct {
+			Alerts struct {
+				MutingRules []MutingRule `json:"mutingRules"`
+			} `json:"alerts"`
+		} `json:"account"`
+	} `json:"actor"`
+}
+
+const (
+	alertsMutingRulesQuery = `query($accountID: Int!) {
+		actor {
+			account(id: $accountID) {
+				alerts {
+					mutingRules {
+						name
+						condition {
+							operator
+							conditions {
+								attribute
+								operator
+								values
+							}
+						}
+					}
+				}
+			}
+		}
+	} `
+
+	alertsMutingRulesCreate = `mutation CreateRule($accountID: Int!, $rule: AlertsMutingRuleInput!) {
+		alertsMutingRuleCreate(accountId: $accountID, rule: $rule) {
+			id
+		}
+	}`
+
+	alertsMutingRulesUpdate = `mutation UpdateRule($accountID: Int!, $ruleID: ID!, $rule: AlertsMutingRuleUpdateInput!) {
+		alertsMutingRuleUpdate(accountId: $accountID, id: $ruleID, rule: $rule) {
+			name
```

Comment:
Thoughts on returning other fields here?

Reply:
Included description and enabled.